persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SignificantlyMutatedGeneSparkRepository.java

```diff
@@ -0,0 +1,80 @@
+package org.cbioportal.persistence.spark;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.cbioportal.model.MutSig;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.SignificantlyMutatedGeneRepository;
+import org.cbioportal.persistence.spark.util.ParquetConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Component
+@Qualifier("significantlyMutatedGeneSparkRepository")
+public class SignificantlyMutatedGeneSparkRepository implements SignificantlyMutatedGeneRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(SignificantlyMutatedGeneSparkRepository.class);
+
+    @Override
+    public List<MutSig> getSignificantlyMutatedGenes(String studyId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        Dataset<Row> mutationDf = spark.read()
+            .parquet(PARQUET_DIR + "/" + studyId + "/" + ParquetConstants.DATA_MUTATIONS);
+
+        // for MSK
+        if (studyId.contains("msk")) {
+            Dataset<Row> mutationsMskcc = spark.read()
+                .parquet(PARQUET_DIR + "/" + studyId + "/" + ParquetConstants.DATA_MUTATIONS_MSKCC);
+            mutationsMskcc = mutationsMskcc.drop("cDNA_change");
+            mutationDf = mutationDf.unionByName(mutationsMskcc);
+        }
+        mutationDf.createOrReplaceTempView("mutation");
+
+        StringBuilder sb = new StringBuilder("select first(Hugo_Symbol),");
+        sb.append("Entrez_Gene_Id, ");
```

Comment:
all appends can go into `StringBuilder` constructor argument

Reply:
yes it can. I broke them off for readability.